application/src/main/java/org/ehrbase/application/abac/CustomMethodSecurityExpressionRoot.java

```diff
@@ -307,39 +355,39 @@ private void templateHandling(
                 String content = "";
                 if (authType.equals(POST)) {
                     // @PostAuthorize gives a ResponseEntity type for "returnObject", so payload is of that type
-                    if (((ResponseEntity<?>) payload).hasBody()) {
-                        Object body = ((ResponseEntity<?>) payload).getBody();
-                        // can have "No content" here (even with some data in the body) if the compo was (logically)
-                        // deleted
-                        if (((ResponseEntity<?>) payload).getStatusCode().equals(HttpStatus.NO_CONTENT)) {
-                            if (body instanceof Map) {
-                                Object error = ((Map<?, ?>) body).get("error");
-                                if (error != null && ((String) error).contains("delet")) {
-                                    // composition was deleted, so nothing to check here, skip
-                                    break;
-                                }
+                    if (!(payload instanceof ResponseEntity<?>)) {
+                        throw new InternalServerException("ABAC: unexpected empty response body");
+                    }
+                    ResponseEntity<?> responseEntity = (ResponseEntity<?>) payload;
```

Comment:
358 -364 are the same error message, put them together

Reply:
I changed the message in other pr